grails-app/controllers/com/careerfly/user/ProfileController.groovy

```diff
@@ -0,0 +1,39 @@
+package com.careerfly.user
+
+class ProfileController {
+
+    def beforeInterceptor = {
+        if(!session.loggedInUser){
+            flash.message= ">>To continue further, You first need to login >>"
+            redirect(controller: "login" , action: "index")
+            return false
+        }
+    }
+    def index() {
+        //User usr=User.findById(session.loggedInUser)
+        [userData : User.get(session.loggedInUser)]
+
+    }
+
+    def update() {
+
+        println "${params}"
+        User updateInstance = User.get(session.loggedInUser)
+        println updateInstance
+        println updateInstance.lastName
+        updateInstance.firstName = params.firstName
+        updateInstance.lastName = params.lastName
+        println params.firstName
+        println updateInstance.firstName
+        updateInstance.email = params.email
+        updateInstance.password = params.password
+        if (updateInstance.hasErrors()) {
+            render(view: 'index', model: [userData: updateInstance])
```

Comment:
There should be a `return` after this line otherwise the lines after this `if` condition will be executed.